### ISSUE TITLE
Fix Swagger paths in 2.3.0

### ISSUE THREAD

**Issue by futurechimp** [scalatra/scalatra#360]
Just to get some docs on the problems we're discussing in IRC, I'll drop what I'm seeing in an issue.

Doing this works:

``` scala
context.mount(new ApiController, "/hackers-api", "hackers-api")
```

Doing any of the following doesn't work:

``` scala
context.mount(new ApiController, "/hackers-api")
// localhost:8080/api-docs/hackers-api.json is a 404

context.mount(new ApiController, "/hackers/api", "hackers/api")
// localhost:8080/api-docs/hackers/api.json is a 404
```

Setting `applicationName` in the controller has no good effect. It does affect the output in localhost:8080/api-docs with whatever `applicationName` is set, but the JSON spec file describing the API still exists wherever it was mounted in `ScalatraBootstrap`, so the `applicationName` change misdirects the docs generator.

Basically, there's no way to set up Swagger-enabled APIs at sub-paths at the moment, and there's a requirement for the `name` parameter when mounting which wasn't previously necessary. I can see the case for `applicationName` but I suspect that some parts of the system aren't taking account of it.
